### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by freeatnet**
This PR adds some smoke tests around pages provided in this example app. Primarily, the tests are aimed at checking that the app still works when upgrading dependencies. Upgrading dependencies can be automated with dependabot.

### CODE REVIEW

**Comment by adamsoffer** [comment]
@freeatnet wow thanks for the PR, this is great! 

I just ran the tests locally and it's failing on my end.
`Cannot find module 'next-server/dist/lib/router-context' from 'index.js'`

Are you able to reproduce?

File: pages/__tests__/about.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/* eslint-env jest */
+
+import React from "react";
+import { act, render } from "@testing-library/react";
+
+import { RouterContext } from "next-server/dist/lib/router-context";
```
Comment (adamsoffer) [inline pages/__tests__/about.js]: path should be `'next/dist/next-server/lib/router-context'`

File: pages/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/* eslint-env jest */
+
+import React from "react";
+import { act, render, waitForElement } from "@testing-library/react";
+
+import { RouterContext } from "next-server/dist/lib/router-context";
```
Comment (adamsoffer) [inline pages/__tests__/index.js]: path should be `'next/dist/next-server/lib/router-context'`

**Comment by freeatnet** [comment]
@adamsoffer right, sorry, I should have clarified that the tests were targetting next 9.0.5 and next-apollo 3.1.4 as a proof of concept for them [breaking](https://circleci.com/gh/freeatnet/next-apollo-example/15?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) when dependabot [proposes an upgrade to next 9.0.6](https://github.com/freeatnet/next-apollo-example/pull/1). I've updated the tests as well as package*.json files to highlight compatible versions.

**Comment by adamsoffer** [comment]
@freeatnet Cool yeah I figured. Thanks so much. This is really awesome.

**Pull request merged**